dev/devicelab/lib/tasks/perf_tests.dart

```diff
@@ -1460,7 +1461,39 @@ class CompileTest {
       case DeviceOperatingSystem.fuchsia:
         throw Exception('Unsupported option for Fuchsia devices');
       case DeviceOperatingSystem.macos:
-        throw Exception('Unsupported option for macOS devices');
+        unawaited(stderr.flush());
+        options.insert(0, 'macos');
+        options.add('--tree-shake-icons');
+        options.add('--split-debug-info=infos/');
+        watch.start();
+        await flutter('build', options: options);
+        watch.stop();
+        final String buildDirectoryPath = path.join(
+          cwd,
+          'build',
+          'macos',
+          'Build',
+          'Products',
+          'Release',
+        );
+        final String? appBundlePath =
+            _findIosAppInBuildDirectory(buildDirectoryPath);
+        if (appBundlePath == null) {
+          throw 'Failed to find app bundle in $buildDirectoryPath';
+        }
+        final Directory appBundle = Directory(appBundlePath);
+        final Archive archive = Archive();
+        await appBundle
+            .list(recursive: true)
+            .where((FileSystemEntity e) => e.runtimeType == File)
+            .map<ArchiveFile>((FileSystemEntity e) {
+              assert(e.runtimeType == File);
+              final File file = e as File;
+              return ArchiveFile(file.path, file.statSync().size, file.readAsBytesSync());
+            })
+            .forEach((ArchiveFile file) => archive.addFile(file));
+        releaseSizeInBytes = TarEncoder().encode(archive).length;
```

Comment:
This is double the size of a downloaded app, because it contains both arm64 and x86.  That's the same reason why `getSizesFromIosApp` thins the app first.

Also, `app_framework_uncompressed_bytes` and `flutter_framework_uncompressed_bytes` will be better to report in order to match iOS, otherwise you are including things like the example app icon size, which doesn't matter.

I'd prefer if iOS and macOS looked exactly the same, except for the logic that actually finds the `.app` directory.  The more we can combine iOS and macOS logic in the tests and tool, the better.

Reply:
https://github.com/flutter/flutter/blob/a6c2e711e455bed6948f4f57620840e71c3d9eba/dev/devicelab/lib/tasks/perf_tests.dart#L1423-L1426

`getSizedFromIosApp` does not appear to thin the app first before assigning a `releaseSizeInBytes`. Am I mistaken?

Reply:
You're right.  I guess the point of that one (since it's an ipa) was to see how big it is to submit to the App Store.  And for iOS that only means arm64 since we recently stopped supporting 32-bit armv7, so the "thinning" isn't even doing anything, so that can be removed.

For macOS we could just build `arm64` and then skip the thinning (done below by setting `FLUTTER_XCODE_ARCHS`, which should get parsed and passed into Xcode).  Or we could do as @cbracken pointed out and just not bother with it and just get the full app size with both archs.

How about something like this (though I didn't run it):
```dart
    switch (deviceOperatingSystem) {
      case DeviceOperatingSystem.ios:
      case DeviceOperatingSystem.macos:
        options.insert(0, deviceOperatingSystem == DeviceOperatingSystem.ios ? 'ios' : 'macos'); // Enhanced enum instead?
        options.add('--tree-shake-icons');
        options.add('--split-debug-info=infos/');
        watch.start();
        await flutter('build', options: options, environment: <String, String>{
          'FLUTTER_XCODE_ARCHS': 'arm64',
        });
        watch.stop();
        final Directory appBuildDirectory = Directory(path.join(cwd, 'build')); // This should be safe since there's a clean right before it.
        final Directory? appBundle = appBuildDirectory
            .listSync()
            .whereType<Directory?>()
            .singleWhere((Directory? directory) =>
              directory != null && path.extension(directory.path) == '.app',
              orElse: () => null);
        if (appBundle == null) {
          throw 'Failed to find app bundle in ${appBuildDirectory.path}';
        }
        final String appPath =  appBundle.path;
        releaseSizeInBytes = ... // Don't `tar`, zip or sum up the files as Chris suggests in his comment.
        if (reportPackageContentSizes) {
          metrics.addAll(await getSizesFromDarwinApp(appPath, deviceOperatingSystem));
        }
        break;
```
```dart
  static Future<Map<String, dynamic>> getSizesFromDarwinApp(String appPath, DeviceOperatingSystem operatingSystem) async {
    final String frameworkDirectory;
    if (operatingSystem == DeviceOperatingSystem.ios) {
      frameworkDirectory = path.join(appPath, 'Frameworks');
    } else {
      // macOS
      frameworkDirectory = path.join(appPath, 'Contents', 'Frameworks');
    }
    final File appFramework = File(path.join(frameworkDirectory, 'App.framework', 'App'));
    final File flutterFramework;

    if (operatingSystem == DeviceOperatingSystem.ios) {
      flutterFramework = File(path.join(frameworkDirectory, 'Flutter.framework', 'Flutter'));
    } else {
      // macOS
      flutterFramework = File(path.join(frameworkDirectory, 'FlutterMacOS.framework', 'FlutterMacOS'));
    }

    return <String, dynamic>{
      'app_framework_uncompressed_bytes': await appFramework.length(),
      'flutter_framework_uncompressed_bytes': await flutterFramework.length(),
    };
  }
```

Reply:
@jmagman this following code should address the feedback above
https://github.com/flutter/flutter/blob/d70df45883b810910e8b90a9e5aec1f750869b7b/dev/devicelab/lib/tasks/perf_tests.dart#L1439-L1479

Thank you for the feedback 👍